### ISSUE TITLE
Don't fall back to CPU if IPUs selected but not available

### ISSUE THREAD

**Issue by callumm-graphcore** [datamol-io/graphium#379]
When I tried to run a Graphium script when IPUs were not available (because I had not set the correct environment variables), I got the following cryptic error message:

```
  File "/localdata/callumm/valence/venv-test-new-lightning/lib/python3.8/site-packages/lightning/pytorch/core/optimizer.py", line 168, in _init_optimizers_and_lr_schedulers
    optim_conf = call._call_lightning_module_hook(model.trainer, "configure_optimizers", pl_module=model)
  File "/localdata/callumm/valence/venv-test-new-lightning/lib/python3.8/site-packages/lightning/pytorch/trainer/call.py", line 140, in _call_lightning_module_hook
    output = fn(*args, **kwargs)
  File "/localdata/callumm/valence/graphium/graphium/trainer/predictor.py", line 210, in configure_optimizers
    optimiser = MuAdam(self.parameters(), **self.optim_options.optim_kwargs, impl=impl)
  File "/localdata/callumm/valence/venv-test-new-lightning/lib/python3.8/site-packages/mup/optim.py", line 70, in MuAdam
    return impl(new_param_groups, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'loss_scaling'
```

This happened because Graphium (via Lightning?) falls back to CPU if IPUs are not available, which I don't think should happen. It would be better if the script crashed when IPUs were not found instead.

**Comment by DomInvivo** [comment]
It's not due to lightning. It happens in the `get_accelerator` function of `_loader.py`. Should be an easy fix

https://github.com/datamol-io/graphium/blob/7b220eb456cedc88fae8373ff3958ed5c05779b0/graphium/config/_loader.py#L47-L57

**Comment by DomInvivo** [comment]
Fixed in #395
